### PR TITLE
Twitch Emote Support

### PR DESCRIPTION
**Pull request by Morgul**
Dead simply emote support. 🎉 

Instead of doing all the crazy looking up the emotes on the fly, blah, blah, oAuth whatever, I decided to go with a brain-dead simple solution. We look up the emotes in a script and dump them to a `json` file as a one time static thing and then just load that into memory in the chat window and do fast lookups on an in-memory dictionary. Then the file can be updated as needed, but emotes rarely change, so treating it as static seems Good Enough™️.

This PR contains two main parts. First is `dump_emotes.py` which will dump emotes into either a combined `emotes.json` or seperate files depending on their source. Currently supports twitch global emotes, twitch channel emotes and BetterTTV emotes. (7TV has no documentation, so that can come later once I rip apart the chrome extension.)

The `emotes.json` file is considered static data to be loaded by the javascript. If, for whatever reason, 'dynamic' emote support is needed, the code in `dump_emotes.py` could be used to build a flask endpoint to build the emote.json file dynamically. (Seems like overkill, imho.)

The second part is just `parser.js`. Originally I wrote this to just parse emotes, but I realized if we ever want to support things (url parsing, or whatever weird things twitch/yt add next) making it a more generic message parser made more sense.

It's written as a JS class with no deps and _should_ be easy to understand/extend, but I'm a JS dev by day, so if I'm getting too fancy, let me know and I'll simplify. I did my best to document everything.

Let me know if you have any question/comments/concerns!

### CODE REVIEW

**Comment by Morgul** [comment]
Just a note about YouTube Emotes: 

They don't currently seem to have a url for getting a list of emotes. Some people have reversed engineered _some_ of the emotes as being available at:

* `https://www.gstatic.com/youtube/img/emojis/emojis-svg-N.json`

with `N` between `0` and `8`. Other people are just raw scraping the YT Chat page.

tl;dr: Sounds like a nightmare I didn't want to mess with.
